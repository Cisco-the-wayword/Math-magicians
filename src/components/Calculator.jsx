import React from 'react';
import Buttons from './Buttons';

export default function Calculator() {
  return (
    <>
      <div className="container">
        <input type="text" className="resultBar" placeholder="0" />
        <Buttons />
      </div>
    </>
  );
}