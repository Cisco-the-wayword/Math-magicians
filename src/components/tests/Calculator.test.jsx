import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../Calculator';

describe('Calculator Page', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should return product of two numbers', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));
    const result = screen.getByRole('none');
    expect(result.textContent).toBe('6');
  });
});