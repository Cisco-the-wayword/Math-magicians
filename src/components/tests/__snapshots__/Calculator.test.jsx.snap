// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Page renders correctly 1`] = `
<DocumentFragment>
  <div
    class="calcPage"
  >
    <h2
      class="calcText"
    >
      Let's solve math problems!
    </h2>
    <div
      class="calculator"
    >
      <div
        class="resultBar"
        role="none"
      >
        0
      </div>
      <div
        class="keypad"
      >
        <button
          class="singleButton"
          type="button"
        >
          AC
        </button>
        <button
          class="singleButton"
          type="button"
        >
          +/-
        </button>
        <button
          class="singleButton"
          type="button"
        >
          %
        </button>
        <button
          class="singleButton"
          type="button"
        >
          รท
        </button>
        <button
          class="singleButton"
          type="button"
        >
          7
        </button>
        <button
          class="singleButton"
          type="button"
        >
          8
        </button>
        <button
          class="singleButton"
          type="button"
        >
          9
        </button>
        <button
          class="singleButton"
          type="button"
        >
          x
        </button>
        <button
          class="singleButton"
          type="button"
        >
          4
        </button>
        <button
          class="singleButton"
          type="button"
        >
          5
        </button>
        <button
          class="singleButton"
          type="button"
        >
          6
        </button>
        <button
          class="singleButton"
          type="button"
        >
          -
        </button>
        <button
          class="singleButton"
          type="button"
        >
          1
        </button>
        <button
          class="singleButton"
          type="button"
        >
          2
        </button>
        <button
          class="singleButton"
          type="button"
        >
          3
        </button>
        <button
          class="singleButton"
          type="button"
        >
          +
        </button>
        <button
          class="doubleButton"
          type="button"
        >
          0
        </button>
        <button
          class="singleButton"
          type="button"
        >
          .
        </button>
        <button
          class="singleButton"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
